Teacher Workout/Features/Discover/Coordinator/DiscoverCoordinator.swift

```diff
@@ -22,8 +22,31 @@ final class DiscoverCoordinator: NSObject, Coordinator {
 extension DiscoverCoordinator: DiscoverViewDelegate {
     func discoverView(_: DiscoverView, didSelectTheme theme: Theme) {
         let viewModel = ThemeDetailsViewModel(theme: theme)
+
+        viewModel.lessonSelected = { [weak self] lesson in
```

Comment:
I am not sure if this is the best place where we should listen for selected lessons. This delegate should only present the ThemeDetails View. Listening to selected lessons and presenting the LessonIntroView shouldn't Theme View do that?

Reply:
Somehow I think each of the coordinators should be in charge of presenting it, but I got your point. 

My problem was that you can get to lesson intro screen from three different scenarios:
1. HomeView -> (select lesson) -> LessonIntroView
2. HomeView -> (select theme) -> ThemeGrid ->  -> LessonIntroView
3. DiscoverView -> (select theme) -> ThemeGrid -> (select lesson) -> LessonIntroView

Wanting to use closures instead of delegates I had to find somewhere a common reference to "listen" to the closure change so that's why I put it there. But I agree it looks weird to have it in a method that does something else.

So I switched back to delegates :)